osu.Game/Screens/Play/HUD/SkinnableInfo.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Extensions;
+using osu.Game.IO.Serialization;
+using osu.Game.Skinning;
+using osuTK;
+
+namespace osu.Game.Screens.Play.HUD
+{
+    /// <summary>
+    /// Serialised information governing custom changes to an <see cref="ISkinSerialisable"/>.
+    /// </summary>
+    [Serializable]
+    public class SkinnableInfo : IJsonSerializable
+    {
+        public Type Type { get; set; }
+
+        public Vector2 Position { get; set; }
+
+        public float Rotation { get; set; }
+
+        public Vector2 Scale { get; set; }
+
+        public Anchor Anchor { get; set; }
+
+        public Anchor Origin { get; set; }
+
+        public List<SkinnableInfo> Children { get; } = new List<SkinnableInfo>();
+
+        public SkinnableInfo()
+        {
+        }
+
+        public SkinnableInfo(Drawable component)
+        {
+            Type = component.GetType();
+
+            Position = component.Position;
+            Rotation = component.Rotation;
+            Scale = component.Scale;
+            Anchor = component.Anchor;
+            Origin = component.Origin;
+
+            if (component is Container container)
```

Comment:
Might be safer as

```csharp
            if (component is IReadOnlyList<Drawable> container)
```

Would be nice to somehow handle `Container<T>` types which use custom drawables, but not sure if that's possible.